backend/app/Listeners/NotifyUsersAboutUpdatedSubmissionStatus.php

```diff
@@ -0,0 +1,226 @@
+            'user' => [
+                'id' => $user->id,
+                'name' => $user->name,
+                'username' => $user->username,
+            ],
+            'publication' => [
+                'id' => $publication->id,
+                'name' => $publication->name,
+            ],
+            'type' => 'submission.updated',
+            'url' => url('/submission/' . $event->submission->id),
+            'action' => 'View Submission',
+        ];
+        $actions = [
+            1 => $this->initiallySubmitted($default),
+            2 => $this->resubmissionRequested($default),
+            3 => $this->resubmitted($default),
+            4 => $this->acceptedForReview($default),
+            5 => $this->rejected($default),
+            6 => $this->acceptedAsFinal($default),
+            7 => $this->expired($default),
+            8 => $this->underReview($default),
+            9 => $this->awaitingDecision($default),
+            10 => $this->awaitingRevision($default),
+            11 => $this->archived($default),
+            12 => $this->deleted($default),
+        ];
+        $notification_data = $actions[$status];
+
+        // Notify submitters, reviewers, review coordinators, and editors
+        Notification::send(
+            $event->submission->users,
+            new SubmissionStatusUpdated($notification_data)
+        );
+        Notification::send($event->submission->publication->editors, new SubmissionStatusUpdated($notification_data));
+    }
+
+    /**
+     * @param array $default
+     * @return array
+     */
+    public function initiallySubmitted($default)
+    {
+        $default['subject'] = 'Submission Submitted';
+        $default['body'] = $default['submission']['title'] . ' has been submitted for review. ' .
+                           'Its fitness for review is to be determined.';
+        $default['type'] = 'submission.initially_submitted';
+
+        return $default;
+    }
+
+    /**
+     * @param array $default
+     * @return array
+     */
+    public function resubmissionRequested($default)
+    {
+        $default['subject'] = 'Resubmission Requested';
+        $default['body'] = $default['submission']['title'] . ' has been determined to be unfit for review. ' .
+                           'A request has been made for resubmission.';
+        $default['type'] = 'submission.resubmission_requested';
+
+        return $default;
+    }
+
+    /**
+     * @param array $default
+     * @return array
+     */
+    public function resubmitted($default)
+    {
+        $default['subject'] = 'Submission Resubmitted';
+        $default['body'] = $default['submission']['title'] . ' has been resubmitted for review.';
+        $default['type'] = 'submission.resubmitted';
+
+        return $default;
+    }
+
+    /**
+     * @param array $default
+     * @return array
+     */
+    public function acceptedForReview($default)
+    {
+        $default['subject'] = 'Submission Accepted for Review';
```

Comment:
Grant, do you think that we should move these to the index.js file? Just a thought!